### PR TITLE
Fix error checks for passive scalar inlet in NSFVAction

### LINKED ISSUE #22774: Error check in NSFVAction prevents the definition of inlet functions for multiple passive salars
## Bug Description
The error check for passive scalar inlet functions in the NSFVAction does not cover the wrong size for each parameter plus checks the wrong dimension on `_passive_scalar_inlet_function`. 
@snschune noticed this while experimenting with an inlet-outlet configuration of an MSR.

## Steps to Reproduce
Take any NSFVAction example with inlet-outlet configuration and try to add more than one passive scalars to the problem.

## Impact
Makes it impossible to simulate multiple passive scalars in an inlet-outlet configuration with the action syntax.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #22774

### CODE REVIEW

**Review by GiudGiud — APPROVED** [review]

**Review by GiudGiud — CHANGES_REQUESTED** [review]
test needs update, currently:
```
navier_stokes/test:finite_volume/ins/action/errors.passive-scalar-inlet-function-error: Size \(0\) is not the same as the number of boundaries in 'passive_scalar_inlet_types' \(size 1\)
```

**Comment by grmnptr** [comment]
Oh yes, I am on it. Just needed to take care of a few things. The change will come in a few mins.

File: modules/navier_stokes/test/tests/finite_volume/ins/action/errors/tests [hunk 1]
```diff
@@ -6,95 +6,117 @@
...
     cli_args = "Modules/NavierStokesFV/passive_scalar_inlet_function=''"
-    expect_err = "Size \(0\) is not the same as the number of boundaries in 'passive_scalar_inlet_types' \(size 1\)"
+    expect_err = "Size \(0\) is not the same as the number of names in 'passive_scalar_names' \(size "
+                 "1\)"
+    ad_indexing_type = 'global'
+  []
+  [passive-scalar-multiple-inlet-function-error]
+    type = 'RunException'
+    input = 2d-rc-error-action.i
+    requirement = 'The system shall throw an error if the number of passive scalar inlet functions does not match the number of inlet boundaries.'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/finite_volume/ins/action/errors/tests:88]: duplicate of the one above

**Review by GiudGiud — APPROVED** [review]

**Comment by grmnptr** [comment]
Thanks @GiudGiud for paying a close attention to this!

**Pull request merged**